src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -1189,6 +1200,11 @@ class run_impl
       : arg_setter(data)
     {}
 
+    virtual void
+    set_sk_args(const uint8_t* args, const uint32_t size)
+    {
+      throw xrt_core::error("Invalid arguments. Reserved for soft kernels");
+    }
```

Comment:
If the base class has multiple virtual functions some of which are not supported by all derived classes, then don't make the pure virtual, instead default their implementation (throw) in the base class.   However I don't think a new function is needed for soft kernels (see other comments)